main/src/ca/uwaterloo/flix/language/phase/Effects.scala

```diff
@@ -577,6 +577,49 @@ object Effects extends Phase[Root, Root] {
           val eff = ast.Eff.Top
           Expression.NativeMethod(method, args, tpe, eff, loc).toSuccess
 
+        /**
+          * New Channel Expression
+          */
+        case Expression.NewChannel(tpe, eff, loc) =>
+          Expression.NewChannel(tpe, eff, loc).toSuccess
+
+        /**
+          * Get Channel Expression
+          */
+        case Expression.GetChannel(exp, tpe, eff, loc) =>
+          for {
+            e <- visitExp(exp, env0)
+          } yield {
+            Expression.GetChannel(e, tpe, eff, loc)
+          }
+
+        /**
+          * Put Channel Expression
+          */
+        case Expression.PutChannel(exp1, exp2, tpe, eff, loc) =>
+          //TODO SJ: Why does Let have "val eff = e1. eff seq e2.eff" - should we do that?
```

Comment:
This class is, unfortunately, not used at  the moment. So anything is really fine. This was one of the other projects :)